File: src/minigames/guess_it/src/main/java/work/lclpnet/ap2/game/guess_it/GuessItMiniGame.java
package work.lclpnet.ap2.game.guess_it;

import net.minecraft.core.RegistryAccess;
import net.minecraft.resources.Identifier;
import net.minecraft.world.item.ItemStack;
import net.minecraft.world.item.Items;
import org.jetbrains.annotations.NotNull;
import work.lclpnet.ap2.ApConstants;
import work.lclpnet.ap2.api.game.GameStartContext;
import work.lclpnet.ap2.api.game.GameType;
import work.lclpnet.ap2.game.MiniGame;
import work.lclpnet.ap2.game.MiniGameFactory;
import work.lclpnet.ap2.game.util.MapLevelGameFactory;

public class GuessItMiniGame implements MiniGame {
    @Override
    public @NotNull Identifier getId() {
        return ApConstants.identifier("guess_it");
    }

    @Override
    public @NotNull GameType getType() {
        return GameType.FFA;
    }

    @Override
    public @NotNull String getAuthor() {
        return ApConstants.PERSON_LCLP;
    }

    @Override
    public @NotNull ItemStack getIcon(@NotNull RegistryAccess manager) {
        return new ItemStack(Items.KNOWLEDGE_BOOK);
    }

    @Override
    public boolean canBeFinale(@NotNull GameStartContext context) {
        return false;  // multiple players can have the same score
    }

    @Override
    public boolean canBePlayed(@NotNull GameStartContext context) {
        return true;
    }

    @Override
    public @NotNull MiniGameFactory createFactory() {
        return new MapLevelGameFactory(GuessItInstance::new);
    }
}

File: src/minigames/guess_it/src/main/kotlin/work/lclpnet/ap2/game/guess_it/GuessItMiniGame.kt
package work.lclpnet.ap2.game.guess_it

import net.minecraft.core.RegistryAccess
import net.minecraft.world.item.ItemStack
import net.minecraft.world.item.Items
import work.lclpnet.ap2.ApConstants
import work.lclpnet.ap2.api.base.GameStartContext
import work.lclpnet.ap2.game.GameType
import work.lclpnet.ap2.game.MiniGame

class GuessItMiniGame : MiniGame {
    override val id = ApConstants.identifier("guess_it")
    override val type = GameType.FFA
    override val author = ApConstants.PERSON_LCLP
    override fun getIcon(manager: RegistryAccess) = ItemStack(Items.KNOWLEDGE_BOOK)
    override fun canBeFinale(context: GameStartContext) = false  // multiple players can have the same score
    override fun canBePlayed(context: GameStartContext) = true
    override fun createFactory() = GuessItFactory()
}
